calculator/main.py

```diff
@@ -37,14 +66,30 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    # first check if we passed the operation as a proper dict.
+    if not isinstance(operation, dict):
+        raise InvalidOperation('Given operation is invalid.')
+
+    # This is a test to see if the op is a valid function
+    # including this requires enabling the test suite:
+    # test_add_new_operations_invalid_operation_type_in_dict
+    '''
+    if not hasattr(list(operation.values())[0], '__call__'):
+        raise InvalidOperation('Given operation is invalid.')
+    '''
+    
+    # Replace operations with new operations dict empty, otherwise update dict.
+    if not calc['operations']:
+        calc['operations'] = operation
+    else:
+        calc['operations'].update(operation)
 
```

Comment:
calc['operations'].update(operation) is enough to do what you want. you dont need to check if dict is empty because the update method will just add the operation in the empty dict